bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -108,10 +108,18 @@ public int size() {
 
     /**
      * Remove an item from the map.
+     *
      * @param key key of the item to be removed
+     * @return boolean to indicate whether given key is removed.
      */
-    public void remove(K key) {
-        map.remove(key);
+    public boolean remove(K key) {
+        boolean hasKey = map.containsKey(key);
+        if (hasKey) {
+            map.remove(key);
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
Can we simplify this to 
`if (hasKey) {
            map.remove(key);
        } 
        return hasKey;
`

Reply:
+1 :)